addons/website_sale/tests/test_website_sale_access.py

```diff
@@ -0,0 +1,33 @@
+from odoo.fields import Command
+from odoo.tests.common import HttpCase, tagged
+
+
+@tagged('post_install', '-at_install')
+class TestWebsiteSaleAccess(HttpCase):
+
+    @classmethod
+    def setUpClass(self):
+        super().setUpClass()
+        self.website = self.env['website'].get_current_website()
+        self.env['ir.ui.view'].search([
+            ('key', '=', 'website_sale.products_categories')
+        ]).active = True
+        self.env['ir.ui.view'].search([
+            ('key', '=', 'website_sale.option_collapse_products_categories')
+        ]).active = False
+
+        self.portal_user = self.env['res.users'].with_context(no_reset_password=True).create({
+            'website_id': self.website.id,
+            'login': 'portal_user',
+            'name': 'Portal User',
+            'group_ids': [Command.link(self.env.ref('base.group_portal').id)],
+        })
+
+    def test_public_and_portal_users_can_access_shop(self):
```

Comment:
The test wasn't immediately clear to me when I first read its name. Could you find a name that better describes it? Here you are testing the shop access whith the public categories restrictions. So maybe...

```suggestion
    def test_public_categories_access_to_portal_users_on_the_shop(self):
```

A little long though 😅